src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -73,10 +72,6 @@ class LogInWithShortLivedTokenPage extends Component {
 
         // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
         const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
-        if (exitTo === ROUTES.WORKSPACE_NEW) {
-            // New workspace creation is handled in AuthScreens, not in its own screen
-            return;
-        }
```

Comment:
Interesting. This code worked at one point - but now doesn't. Any ideas about why?

Reply:
I'm not sure why this used to work, maybe a change in react-navigation? 

Anyways, some context here ... in [this PR](https://github.com/Expensify/App/pull/5706/files) I removed the `workspace/new` route. It's not a real route anymore, and that constant is really only used as a flag/url param in `AuthScreens` – if the app was opened by linking to `transition?exitTo="workspace/new"`, then we create a new free policy.

IIRC, the transition page will authenticate us if necessary, then remount on the AuthStack. When that mounts, we check the initial URL and maybe create a new free policy [here](https://github.com/Expensify/App/blob/d13ac5e05d8a923009ef2352dda79df75b373ad6/src/libs/Navigation/AppNavigator/AuthScreens.js#L131-L139). I think I added this code in `componentDidUpdate`:

```js
if (exitTo === ROUTES.WORKSPACE_NEW) {
    // New workspace creation is handled in AuthScreens, not in its own screen
    return;
}
```

Thinking something to the effect of "this route doesn't actually exist, so we shouldn't try to route there". And that still makes sense, but this lands us on the transition page forever. In effect, this PR links to a route that doesn't exist, and then react-navigation defaults to home. So we should probably just make that explicit and navigate to home if our `exitTo` is `workspace/new`.

What's weird is I don't remember this being an issue when I tested this flow many times 😕 :fry:

Reply:
chatting with @ctkochan22 we figured it out by looking at the git history for `Policy.js`

Reply:
Okay, that was super informative. 

With that in mind, I think I figured out the right fix. https://github.com/Expensify/Expensify/issues/188048#issuecomment-995346509